orca-webhook/src/main/java/com/netflix/spinnaker/orca/webhook/tasks/MonitorWebhookTask.java

```diff
@@ -0,0 +1,270 @@
+      }
+
+      String errorMessage =
+          String.format(
+              "an exception occurred in webhook monitor to %s: %s", stageData.statusEndpoint, e);
+      log.error(errorMessage, e);
+      monitor.put("error", errorMessage);
+      return TaskResult.builder(ExecutionStatus.TERMINAL).context(originalResponse).build();
+    }
+
+    var responsePayload = originalResponse;
+    monitor.putAll(
+        Map.of(
+            "body", response.getBody(),
+            "statusCode", response.getStatusCode(),
+            "statusCodeValue", response.getStatusCode().value()));
+
+    if (!response.getHeaders().isEmpty()) {
+      monitor.put("headers", response.getHeaders().toSingleValueMap());
+    }
+
+    if (Strings.isNullOrEmpty(stageData.statusJsonPath)) {
+      return TaskResult.builder(ExecutionStatus.SUCCEEDED).context(originalResponse).build();
+    }
+
+    Object result;
+    try {
+      result = JsonPath.read(response.getBody(), stageData.statusJsonPath);
+    } catch (PathNotFoundException e) {
+      monitor.put(
+          "error", String.format(JSON_PATH_NOT_FOUND_ERR_FMT, "status", stageData.statusJsonPath));
+      return TaskResult.builder(ExecutionStatus.TERMINAL).context(originalResponse).build();
+    }
+    if (!(result instanceof String || result instanceof Number || result instanceof Boolean)) {
+      monitor.putAll(
+          Map.of(
+              "error",
+              String.format(
+                  "The json path '%s' did not resolve to a single value", stageData.statusJsonPath),
+              "resolvedValue",
+              result));
+      return TaskResult.builder(ExecutionStatus.TERMINAL).context(originalResponse).build();
+    }
+
+    if (StringUtils.isNotEmpty(stageData.progressJsonPath)) {
+      Object progress;
+      try {
+        progress = JsonPath.read(response.getBody(), stageData.progressJsonPath);
+      } catch (PathNotFoundException e) {
+        monitor.put(
+            "error",
+            String.format(JSON_PATH_NOT_FOUND_ERR_FMT, "progress", stageData.statusJsonPath));
+        return TaskResult.builder(ExecutionStatus.TERMINAL).context(responsePayload).build();
+      }
+      if (!(progress instanceof String)) {
+        monitor.put(
+            "error",
+            String.format(
+                "The json path '%s' did not resolve to a String value",
+                stageData.progressJsonPath));
+        monitor.put("resolvedValue", progress);
+        return TaskResult.builder(ExecutionStatus.TERMINAL).context(responsePayload).build();
+      }
+      if (progress != null) {
+        monitor.put("progressMessage", progress);
+      }
+    }
+
+    var statusMap =
+        createStatusMap(
+            stageData.successStatuses, stageData.canceledStatuses, stageData.terminalStatuses);
+
+    if (result instanceof Number) {
+      var status = result.equals(100) ? ExecutionStatus.SUCCEEDED : ExecutionStatus.RUNNING;
```

Comment:
also TIL (you can't use a numeric value as an indication of success or failure, it has to be a percentage completion...) 

(again not suggesting a change here...)